pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -1796,6 +1797,264 @@ func TestReconciler(t *testing.T) {
 			},
 			workloadCmpOpts: defaultWorkloadCmpOpts,
 		},
+		"excess pods before wl creation, youngest pods are deleted": {
+			pods: []corev1.Pod{
+				*basePodWrapper.
+					Clone().
+					Label("kueue.x-k8s.io/managed", "true").
+					KueueFinalizer().
+					KueueSchedulingGate().
+					Group("test-group").
+					GroupTotalCount("1").
+					CreationTimestamp(time.Date(2022, 9, 15, 11, 0, 0, 0, time.UTC)).
```

Comment:
a strategy we use in other places is to have a variable `now := time.Now()` at the beginning of the test, and then have calls like: `CreationTimestamp(now.Add(time.Second))`, which would read as "1 second ago".